glib-macros/src/downgrade_enum.rs

```diff
@@ -0,0 +1,74 @@
+use crate::downgrade_fields::{derive_downgrade_fields, DowngradeStructParts};
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::Ident;
+
+pub fn derive_downgrade_for_enum(ident: Ident, data_enum: syn::DataEnum) -> TokenStream {
```

Comment:
Just thought about this but could you at least put some small comments on each function explaining what it's doing please?

Reply:
Done